<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Gillespie Simulation for Break-Fusion-Bridge (BFB) Processes Modified to support diploid chromosomes (alleles A and B) — bridge_sim • bridges</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gillespie Simulation for Break-Fusion-Bridge (BFB) Processes Modified to support diploid chromosomes (alleles A and B) — bridge_sim"><meta name="description" content="Simulates the evolution of cells undergoing Break-Fusion-Bridge cycles using
a continuous-time Gillespie algorithm. This function models cell birth and death
processes with the possibility of BFB events occurring during replication.
Cells with amplified hotspots have an increased birth rate.
Now supports modeling both alleles (A and B) of each chromosome."><meta property="og:description" content="Simulates the evolution of cells undergoing Break-Fusion-Bridge cycles using
a continuous-time Gillespie algorithm. This function models cell birth and death
processes with the possibility of BFB events occurring during replication.
Cells with amplified hotspots have an increased birth rate.
Now supports modeling both alleles (A and B) of each chromosome."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">bridges</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/bridges.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Gillespie Simulation for Break-Fusion-Bridge (BFB) Processes Modified to support diploid chromosomes (alleles A and B)</h1>

      <div class="d-none name"><code>bridge_sim.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Simulates the evolution of cells undergoing Break-Fusion-Bridge cycles using
a continuous-time Gillespie algorithm. This function models cell birth and death
processes with the possibility of BFB events occurring during replication.
Cells with amplified hotspots have an increased birth rate.
Now supports modeling both alleles (A and B) of each chromosome.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bridge_sim</span><span class="op">(</span></span>
<span>  initial_cells <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">22</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span>,</span>
<span>  bin_length <span class="op">=</span> <span class="fl">1e+06</span>,</span>
<span>  birth_rate <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  death_rate <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  bfb_allele <span class="op">=</span> <span class="st">"1:A"</span>,</span>
<span>  normal_dup_rate <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  bfb_prob <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  amp_rate <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  del_rate <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  wgd_available <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  wgd_probability <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  lambda <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  rate <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  positive_selection_rate <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  negative_selection_rate <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  max_time <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  max_cells <span class="op">=</span> <span class="fl">256</span>,</span>
<span>  first_round_of_bfb <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  breakpoint_support <span class="op">=</span> <span class="st">"uniform"</span>,</span>
<span>  hotspot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">"1:A"</span>, pos <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>,</span>
<span>  alpha <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  beta <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  custom_breakpoints <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-initial-cells">initial_cells<a class="anchor" aria-label="anchor" href="#arg-initial-cells"></a></dt>
<dd><p>Numeric. Number of cells at the start of simulation. Default: 1</p></dd>


<dt id="arg-chromosomes">chromosomes<a class="anchor" aria-label="anchor" href="#arg-chromosomes"></a></dt>
<dd><p>Character vector. Chromosomes to model (e.g., c("1", "2", "X")). Default: c(1:22, "X", "Y")</p></dd>


<dt id="arg-bin-length">bin_length<a class="anchor" aria-label="anchor" href="#arg-bin-length"></a></dt>
<dd><p>Numeric. Length of each genomic bin in base pairs. Default: 5e5</p></dd>


<dt id="arg-birth-rate">birth_rate<a class="anchor" aria-label="anchor" href="#arg-birth-rate"></a></dt>
<dd><p>Numeric. Base rate at which cells replicate per unit time. Default: 0.1</p></dd>


<dt id="arg-death-rate">death_rate<a class="anchor" aria-label="anchor" href="#arg-death-rate"></a></dt>
<dd><p>Numeric. Rate at which cells die per unit time. Default: 0.001</p></dd>


<dt id="arg-bfb-allele">bfb_allele<a class="anchor" aria-label="anchor" href="#arg-bfb-allele"></a></dt>
<dd><p>Character. Allele which will be affected by BFB events. Defaul : "1:A"</p></dd>


<dt id="arg-normal-dup-rate">normal_dup_rate<a class="anchor" aria-label="anchor" href="#arg-normal-dup-rate"></a></dt>
<dd><p>Numeric. Rate of normal duplication events. Default: 0.5</p></dd>


<dt id="arg-bfb-prob">bfb_prob<a class="anchor" aria-label="anchor" href="#arg-bfb-prob"></a></dt>
<dd><p>Numeric. Probability of BFB event occurring during replication. Default: 0.01</p></dd>


<dt id="arg-amp-rate">amp_rate<a class="anchor" aria-label="anchor" href="#arg-amp-rate"></a></dt>
<dd><p>Numeric. Rate of amplification events. Default: 0.1</p></dd>


<dt id="arg-del-rate">del_rate<a class="anchor" aria-label="anchor" href="#arg-del-rate"></a></dt>
<dd><p>Numeric. Rate of deletion events. Default: 0.1</p></dd>


<dt id="arg-wgd-available">wgd_available<a class="anchor" aria-label="anchor" href="#arg-wgd-available"></a></dt>
<dd><p>Numeric. Number of whole-genome duplication events available. Default: 0</p></dd>


<dt id="arg-wgd-probability">wgd_probability<a class="anchor" aria-label="anchor" href="#arg-wgd-probability"></a></dt>
<dd><p>Numeric. Probability of whole-genome duplication event occurring. Default: 0.05</p></dd>


<dt id="arg-lambda">lambda<a class="anchor" aria-label="anchor" href="#arg-lambda"></a></dt>
<dd><p>Rate parameter for Poisson distribution used to sample
the number of genomic events per daughter cell</p></dd>


<dt id="arg-rate">rate<a class="anchor" aria-label="anchor" href="#arg-rate"></a></dt>
<dd><p>Rate parameter used in amplification/deletion simulations.
Length of event is sample from exponential distribution with parameter 1 / rate.</p></dd>


<dt id="arg-positive-selection-rate">positive_selection_rate<a class="anchor" aria-label="anchor" href="#arg-positive-selection-rate"></a></dt>
<dd><p>Numeric. Selection advantage for cells with amplified hotspot. Default: 0</p></dd>


<dt id="arg-negative-selection-rate">negative_selection_rate<a class="anchor" aria-label="anchor" href="#arg-negative-selection-rate"></a></dt>
<dd><p>Numeric. Selection disadvantage for cells without amplified hotspot. Default: 0</p></dd>


<dt id="arg-max-time">max_time<a class="anchor" aria-label="anchor" href="#arg-max-time"></a></dt>
<dd><p>Numeric. Maximum simulation time. Default: 50</p></dd>


<dt id="arg-max-cells">max_cells<a class="anchor" aria-label="anchor" href="#arg-max-cells"></a></dt>
<dd><p>Numeric. Maximum number of cells allowed before simulation stops. Default: 100</p></dd>


<dt id="arg-first-round-of-bfb">first_round_of_bfb<a class="anchor" aria-label="anchor" href="#arg-first-round-of-bfb"></a></dt>
<dd><p>Logical. Whether to apply BFB to initial cells. Default: TRUE</p></dd>


<dt id="arg-breakpoint-support">breakpoint_support<a class="anchor" aria-label="anchor" href="#arg-breakpoint-support"></a></dt>
<dd><p>Character. Distribution used for breakpoint
selection ("uniform", "beta", etc.). Default: "uniform"</p></dd>


<dt id="arg-hotspot">hotspot<a class="anchor" aria-label="anchor" href="#arg-hotspot"></a></dt>
<dd><p>Named list. Hotspot positions for each chromosome allele
(e.g., list(chr = "1:A", pos = 100), which is default)</p></dd>


<dt id="arg-alpha">alpha<a class="anchor" aria-label="anchor" href="#arg-alpha"></a></dt>
<dd><p>Numeric. First parameter for beta distribution if used for
breakpoint selection. Default: NULL</p></dd>


<dt id="arg-beta">beta<a class="anchor" aria-label="anchor" href="#arg-beta"></a></dt>
<dd><p>Numeric. Second parameter for beta distribution if used for
breakpoint selection. Default: NULL</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing:</p>
<dl><dt>cells</dt>
<dd><p>List of cell chromosome sequences at the end of simulation</p></dd>

<dt>cell_history</dt>
<dd><p>Data frame with cell birth/death times and lineage information</p></dd>

<dt>input_parameters</dt>
<dd><p>List of input parameters used in the simulation</p></dd>

</dl></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Giovanni Santacatterina.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

