<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Using bridges • bridges</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using bridges">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">bridges</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/bridges.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Using bridges</h1>
            
      

      <div class="d-none name"><code>bridges.Rmd</code></div>
    </div>

    
    
<p>This tutorial demonstrates how to simulate copy number evolution with
BFB (Breakage-Fusion-Bridge) events, infer phylogenies from
allele-specific CNAs, and visualize the output using the
<code>bridges</code> R package.</p>
<div class="section level2">
<h2 id="load-required-libraries">Load Required Libraries<a class="anchor" aria-label="anchor" href="#load-required-libraries"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jovoni.github.io/bridges/">bridges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KlausVigo/phangorn" class="external-link">phangorn</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-1-simulate-copy-number-evolution">Step 1: Simulate Copy Number Evolution<a class="anchor" aria-label="anchor" href="#step-1-simulate-copy-number-evolution"></a>
</h2>
<p>We simulate a single BFB event on chromosome 8, allele A, across 128
cells.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alleles_to_use</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu">bridges</span><span class="fu">::</span><span class="fu"><a href="../reference/bridge_sim.html">bridge_sim</a></span><span class="op">(</span></span>
<span>  max_cells <span class="op">=</span> <span class="fl">128</span>,</span>
<span>  bfb_allele <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"8:A"</span><span class="op">)</span>,</span>
<span>  normal_dup_rate <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  lambda <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sim</span><span class="op">$</span><span class="va">cna_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;   cell_id  chr   bin_idx     A     B    CN   start     end</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> cell_101 1           1     1     1     2       1 1<span style="text-decoration: underline;">000</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> cell_101 1           2     1     1     2 1<span style="text-decoration: underline;">000</span>001 2<span style="text-decoration: underline;">000</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> cell_101 1           3     1     1     2 2<span style="text-decoration: underline;">000</span>001 3<span style="text-decoration: underline;">000</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> cell_101 1           4     1     1     2 3<span style="text-decoration: underline;">000</span>001 4<span style="text-decoration: underline;">000</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> cell_101 1           5     1     1     2 4<span style="text-decoration: underline;">000</span>001 5<span style="text-decoration: underline;">000</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> cell_101 1           6     1     1     2 5<span style="text-decoration: underline;">000</span>001 6<span style="text-decoration: underline;">000</span>000</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-2-fit-the-phylogeny-to-the-cna-data">Step 2: Fit the Phylogeny to the CNA Data<a class="anchor" aria-label="anchor" href="#step-2-fit-the-phylogeny-to-the-cna-data"></a>
</h2>
<p>We run inference on the simulated data, disabling jitter
smoothing.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">k_jitter_fix</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">bridges</span><span class="fu">::</span><span class="fu"><a href="../reference/fit.html">fit</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">sim</span><span class="op">$</span><span class="va">cna_data</span>,</span>
<span>  alleles <span class="op">=</span> <span class="va">alleles_to_use</span>,</span>
<span>  k_jitter_fix <span class="op">=</span> <span class="va">k_jitter_fix</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-3-compare-inferred-and-true-trees">Step 3: Compare Inferred and True Trees<a class="anchor" aria-label="anchor" href="#step-3-compare-inferred-and-true-trees"></a>
</h2>
<p>We compare the inferred tree to the true simulated one using
Robinson-Foulds distance.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">true_tree</span> <span class="op">&lt;-</span> <span class="va">sim</span><span class="op">$</span><span class="va">tree</span></span>
<span><span class="va">inferred_tree</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">tree</span></span>
<span></span>
<span><span class="fu">phangorn</span><span class="fu">::</span><span class="fu"><a href="https://klausvigo.github.io/phangorn/reference/treedist.html" class="external-link">RF.dist</a></span><span class="op">(</span><span class="va">true_tree</span>, <span class="va">inferred_tree</span>, normalize <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.136</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-4-visualize-cn-profiles-with-trees">Step 4: Visualize CN Profiles with Trees<a class="anchor" aria-label="anchor" href="#step-4-visualize-cn-profiles-with-trees"></a>
</h2>
<p>We can visualize both the true and inferred trees alongside
allele-specific CN profiles.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">bridges</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">sim</span><span class="op">$</span><span class="va">cna_data</span>,</span>
<span>  tree <span class="op">=</span> <span class="va">sim</span><span class="op">$</span><span class="va">tree</span>,</span>
<span>  use_raster <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ladderize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  to_plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CN"</span>, <span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>,</span>
<span>  branch_length <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="bridges_files/figure-html/plot-true-tree-1.png" width="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">bridges</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">sim</span><span class="op">$</span><span class="va">cna_data</span>,</span>
<span>  tree <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">tree</span>,</span>
<span>  use_raster <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ladderize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  to_plot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CN"</span>, <span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>,</span>
<span>  branch_length <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="bridges_files/figure-html/plot-inferred-tree-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="step-5-detect-bfb-signatures">Step 5: Detect BFB Signatures<a class="anchor" aria-label="anchor" href="#step-5-detect-bfb-signatures"></a>
</h2>
<p>We use the built-in BFB detection function to quantify “BFB-ness” per
chromosome and allele.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bfb_detection_df</span> <span class="op">&lt;-</span> <span class="fu">bridges</span><span class="fu">::</span><span class="fu"><a href="../reference/detect_bfb.html">detect_bfb</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bfb_detection_df</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 8</span></span></span>
<span><span class="co">#&gt;    mean N_trials N_successes chr   allele p.value threshold adj.pval</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>     0      127           0 1     A            1     0.005        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>     0      127           0 2     A            1     0.005        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span>     0      127           0 3     A            1     0.005        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span>     0      127           0 4     A            1     0.005        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span>     0      127           0 5     A            1     0.005        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span>     0      127           0 6     A            1     0.005        1</span></span>
<span></span>
<span><span class="va">bfb_detection_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">chr</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">22</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">chr</span>, y <span class="op">=</span> <span class="va">mean</span>, fill <span class="op">=</span> <span class="va">allele</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">lims</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Chromosome"</span>, y <span class="op">=</span> <span class="st">"BFB-ness"</span>, fill <span class="op">=</span> <span class="st">"Allele"</span><span class="op">)</span></span></code></pre></div>
<p><img src="bridges_files/figure-html/detect-bfb-1.png" width="100%" style="display: block; margin: auto;"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Giovanni Santacatterina.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
